psh/pm/pm.c

```diff
@@ -119,38 +175,63 @@ int psh_pm(int argc, char *argv[])
 	for (;;) {
 		sleep(interval);
 
-		ctcnt = psh_pm_getThreads(&ctinfo, &ctsize);
-		if (ctcnt < 0)
+		int ctcnt = psh_pm_getThreads(&ctinfo, &ctsize);
+		if (ctcnt < 0) {
+			if ((rebootNoMem != 0) && (ctcnt == -ENOMEM)) {
+				psh_pm_reboot("ENOMEM while getting thread info");
+			}
 			continue;
+		}
 
-		i = 0;
-		j = 0;
+		uint32_t currKernel = 0;
+
+		int i = 0;
+		int j = 0;
 
 		while (i < itcnt) {
-			ipid = itinfo[i].pid;
-			cpid = ctinfo[j].pid;
+			pid_t ipid = itinfo[i].pid;
+			pid_t cpid = ctinfo[j].pid;
+
+			if (ctinfo[i].pid == 0) { /* kernel idle process - shows kernel RAM usage */
+				currKernel = ctinfo[i].vmem;
+			}
 
 			if ((!ignore_ppid || ipid != ppid) && (ipid < cpid)) {
 				fprintf(stderr, "pm: process %d died\n", ipid);
 				if (restart) {
-					err = reboot(PHOENIX_REBOOT_MAGIC);
-					if (err < 0)
-						fprintf(stderr, "pm: failed to restart the machine\n");
+					psh_pm_reboot("monitored process died");
 				}
 			}
 
 			if (ipid <= cpid) {
+				int k;
 				for (k = i + 1; k < itcnt && itinfo[k].pid == itinfo[i].pid; k++)
 					;
 				i = k;
 			}
 
 			if (ipid >= cpid) {
+				int k;
 				for (k = j + 1; k < ctcnt && ctinfo[k].pid == ctinfo[j].pid; k++)
 					;
 				j = k;
 			}
 		}
+
+		if (rebootNoMem != 0) {
+			uint32_t currTotal = psh_pm_getTotal();
```

Comment:
Only if `-m` option was passed.

Reply:
always, I want it for `printf`  in line 224 (to monitor eg. if total ram usage increases in pair with kernel ram usage)